modules/platform/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/CopyDocumentTest.java

```diff
@@ -129,4 +130,27 @@ public void testLifeCycleResetOnCopy() throws OperationException {
             assertEquals(initialLifeCycleState, result.getCurrentLifeCycleState());
         }
     }
+
+    @Test
+    public void testVersionNotNullOnCopy() throws OperationException {
+        DocumentModel source = session.createDocumentModel(ROOT, "Source", FILE);
+        source = session.createDocument(source);
+        source = session.getDocument(session.checkIn(source.getRef(), VersioningOption.MAJOR, ""));
```

Comment:
You could have an extra line for readability/debugging purpose:
```
DocumentRef sourceRef = session.checkIn(source.getRef(), VersioningOption.MAJOR, null);
```
Also, I would pass `null` or a non-empty comment as check-in comment.